src/libs/ck-libs/io/ckio.C

```diff
@@ -398,6 +524,57 @@ namespace Ck { namespace IO {
           return 0;
         }
       };
+
+
+    // class that is used to aggregate the data for a specific read call made by the user
+    class ReadAssembler : public CBase_ReadAssembler {
+	private:
+		std::vector<char> _data_buffer; // the data buffer that is used to store the read request
+		Session _session;
+		size_t _bytes_left; // the number of bytes the user wants for a particular read
+		size_t _read_offset; // the offset they specify for their read
+		CkCallback _after_read; // the callback to invoke after the read is complete
+	public:
+		ReadAssembler(Session session, size_t bytes, size_t offset, CkCallback after_read){
+			_session = session;
+			_bytes_left = bytes;
+			_read_offset = offset;
+			_after_read = after_read;
+			_data_buffer.resize(_bytes_left, 'r'); // resize the buffer to the size of read call
+			_data_buffer.shrink_to_fit();
+			// ckout << "RA Buffer init\n"; 
+			/*
+			for(char ch : _data_buffer)
+				ckout << ch;
+			ckout << endl;
+			std::cout << "ReadAssembler built\n";
+			*/
+		}
+
+		void shareData(size_t read_chare_offset, std::vector<char> data){
```

Comment:
Do not use `std::vector<char>` as an entry method parameter. Change this to `char* data, int size`. You can call `data.data(), data.size()` as parameters when calling the function.